src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Interop/UnsafeNativeMethodsWindow.cs

```diff
@@ -0,0 +1,344 @@
+        }
+
+        var pvAttribute = 0x0;
+
+        _ = Dwmapi.DwmGetWindowAttribute(
+            handle,
+            Dwmapi.DWMWINDOWATTRIBUTE.DWMWA_MICA_EFFECT,
+            ref pvAttribute,
+            Marshal.SizeOf(typeof(int))
+        );
+
+        return pvAttribute == 0x1;
+    }
+
+    /// <summary>
+    /// Tries to apply legacy Mica effect for the selected <see cref="Window"/>.
+    /// </summary>
+    /// <param name="window">The window to which the effect is to be applied.</param>
+    /// <returns><see langword="true"/> if invocation of native Windows function succeeds.</returns>
+    public static bool ApplyWindowLegacyMicaEffect(Window window) =>
+        GetHandle(window, out IntPtr windowHandle) && ApplyWindowLegacyMicaEffect(windowHandle);
+
+    /// <summary>
+    /// Tries to apply legacy Mica effect for the selected <see cref="Window"/>.
+    /// </summary>
+    /// <param name="handle">Window handle.</param>
+    /// <returns><see langword="true"/> if invocation of native Windows function succeeds.</returns>
+    public static bool ApplyWindowLegacyMicaEffect(IntPtr handle)
+    {
+        var backdropPvAttribute = 0x1; //Enable
+
+        _ = Dwmapi.DwmSetWindowAttribute(
+            handle,
+            Dwmapi.DWMWINDOWATTRIBUTE.DWMWA_MICA_EFFECT,
+            ref backdropPvAttribute,
+            Marshal.SizeOf(typeof(int))
+        );
+
+        return true;
+    }
+
+    /// <summary>
+    /// Tries to apply legacy Acrylic effect for the selected <see cref="Window"/>.
+    /// </summary>
+    /// <param name="window">The window to which the effect is to be applied.</param>
+    /// <returns><see langword="true"/> if invocation of native Windows function succeeds.</returns>
+    public static bool ApplyWindowLegacyAcrylicEffect(Window window) =>
+        GetHandle(window, out IntPtr windowHandle) && ApplyWindowLegacyAcrylicEffect(windowHandle);
+
+    /// <summary>
+    /// Tries to apply legacy Acrylic effect for the selected <see cref="Window"/>.
+    /// </summary>
+    /// <param name="handle">Window handle</param>
+    /// <returns><see langword="true"/> if invocation of native Windows function succeeds.</returns>
+    public static bool ApplyWindowLegacyAcrylicEffect(IntPtr handle)
+    {
+        var accentPolicy = new ACCENT_POLICY
+        {
+            nAccentState = ACCENT_STATE.ACCENT_ENABLE_ACRYLICBLURBEHIND,
+            nColor = 0x990000 & 0xFFFFFF
+        };
+
+        var accentStructSize = Marshal.SizeOf(accentPolicy);
+        var accentPtr = Marshal.AllocHGlobal(accentStructSize);
+
+        Marshal.StructureToPtr(accentPolicy, accentPtr, false);
+
+        var data = new WINCOMPATTRDATA
+        {
+            Attribute = WCA.WCA_ACCENT_POLICY,
+            SizeOfData = accentStructSize,
+            Data = accentPtr
+        };
+
+        _ = NativeMethods.SetWindowCompositionAttribute(handle, ref data);
+
+        Marshal.FreeHGlobal(accentPtr);
+
+        return true;
```

Comment:
All this accent policy code appears to be identical to my code distributed via samples and other apps. I don't mind, but this code needs to be very careful scrutinized for applicability to a wide WPF audience!